src/input/InputPlugin.cxx

```diff
@@ -32,14 +32,10 @@ InputPlugin::SupportsUri(const char *uri) const noexcept
 		for (auto i = prefixes; *i != nullptr; ++i)
 			if (StringStartsWithIgnoreCase(uri, *i))
 				return true;
-	} else {
-		for (const auto& schema : protocols()) {
-			if (StringStartsWithIgnoreCase(uri, schema.c_str())){
-				return true;
-			}
-		}
 	}
-	return false;
+
+	return std::any_of(protocols().begin(), protocols().end(), [uri](const auto &schema)
+			{ return StringStartsWithIgnoreCase(uri, schema.c_str()); } );
```

Comment:
You've moved this out of the `else` branch, why?

Reply:
It's basically doing else after return originally. This simplifies it.

Reply:
Before, it never looped over `protocols` if `prefixes!=nullptr`.
With your patch, it always loops over `protocols`, even if `prefixes` has already been checked.
This changes the code's behavior, and I asked "why?" - you may have had reasons to change the code (but it would be a bad idea to do so in a pure refactoring patch) - but you didn't answer my question. I think you weren'd aware of this change.

Reply:
Added a return false statement.